user_tools/src/spark_rapids_pytools/cloud_api/emr.py

```diff
@@ -69,10 +70,19 @@ def _construct_cli_object(self) -> CMDDriverBase:
     def _install_storage_driver(self):
         self.storage = S3StorageDriver(self.cli)
 
-    def _construct_cluster_from_props(self,
-                                      cluster: str,
-                                      props: str = None):
-        return EMRCluster(self).set_connection(cluster_id=cluster, props=props)
+    def _construct_cluster_from_props(self, cluster: str, props: str = None, is_inferred: bool = False):
+        return EMRCluster(self, is_inferred=is_inferred).set_connection(cluster_id=cluster, props=props)
+
+    def _construct_cluster_config(self, cluster_info: dict, default_config: dict):
+        cluster_conf = default_config
+        for group in cluster_conf['Cluster']['InstanceGroups']:
+            if group.get('Name') == 'CORE':
+                group['InstanceType'] = cluster_info['executor_instance']
```

Comment:
we may consider changing the name in scala to instanceType instead of executorType.

Reply:
`instanceType` is a field specific to EMR. Every CSP has a different naming convention.